airflow/decorators/__init__.pyi

```diff
@@ -106,6 +106,11 @@ class TaskDecoratorCollection:
         python_version: None | str | int | float = None,
         use_dill: bool = False,
         system_site_packages: bool = True,
+        pip_install_options: list[str] | None = None,
+        skip_on_exit_code: int | Container[int] | None = None,
+        extra_index_urls: None | Iterable[str] | str = None,
```

Comment:
I would prefer this to be able to override the default index URL, preferrably forcing the user to do so when specifying additional indexes. [Using `--extra-index-url` without also changing the default index URL is a common source of bugs.](https://kambojrehaan.medium.com/dependency-confusion-how-i-hacked-into-apple-microsoft-and-dozens-of-other-companies-e5503bbc97f9#70c6)

I would advise changing this argument to `index_urls: Collection[str] | None = None` and change the implementation so if `index_urls` is not None, the first entry is passed to `--index-url`, and from the second on to `--extra-index-url`. A nice addition would be to padd `--no-index-url` if an empty list is provided.

Reply:
No long thought - convinced! Thanks for the feedback. That totally makes sense. Great idea and much more versatile than initial proposal.